build/Targets/Settings.props

```diff
@@ -239,6 +239,7 @@
     <DefineConstants Condition="'$(InitialDefineConstants)' != ''">$(InitialDefineConstants)</DefineConstants>
 
     <!-- The NoWarn property is still being unconditionally set.  https://github.com/dotnet/sdk/issues/752 -->
-    <NoWarn>$(NoWarn);1591</NoWarn>
+    <OptionStrict>On</OptionStrict>
+    <NoWarn>$(NoWarn);1591;41999;42016</NoWarn>
```

Comment:
Just saw this.  We should not need these suppressions now.  Turning on OptionStrict should elimanet the warnings because the compiler will pick the correct overload.